okhttp-logging-interceptor/src/main/java/okhttp3/logging/HttpLoggingInterceptor.java

```diff
@@ -251,6 +259,27 @@ public Level getLevel() {
     return response;
   }
 
+  private boolean isPlainText(Buffer buffer) throws IOException {
+    Buffer out = new Buffer();
+    long bytes = buffer.size() < 64 ? buffer.size() : 64;
+    buffer.copyTo(out, 0, bytes);
+    int charCount = 0;
+    for (int i = 0; i < 16; i++) {
+      if (out.exhausted()) {
+        break;
+      }
+      if (isNonPrintableCharacter(out.readUtf8CodePoint())) {
+        charCount++;
+      }
+    }
+    return charCount < 2;
+  }
+
+  private boolean isNonPrintableCharacter(int codePoint) {
+    return codePoint == '\ufffd' || ('\u0000' <= codePoint && codePoint <= '\u001f')
```

Comment:
I agree with you. I think it's better to err on the side of logging the body. I included it because I noticed error conditions in `readUtf8CodePoint()` return `\ufffd`.